src/sidebar/services/auth.js

```diff
@@ -142,45 +124,91 @@ export class AuthService extends TinyEmitter {
       $window.addEventListener('storage', ({ key }) => {
         if (key === storageKey()) {
           // Reset cached token information. Tokens will be reloaded from storage
-          // on the next call to `tokenGetter()`.
+          // on the next call to `getAccessToken()`.
           tokenInfoPromise = null;
           this.emit('oauthTokensChanged');
         }
       });
     };
 
-    function oauthClient() {
+    const oauthClient = async () => {
```

Comment:
Using an arrow function here will avoid a hazard with `this` in a later refactoring.

Reply:
I noticed that on the `login` method below you chose not to use arrow function.

Reply:
It's more a case of "I didn't get to refactoring it in this PR".